site-src/geps/gep-724.md

```diff
@@ -0,0 +1,624 @@
+	//
+	// Support: Core
+  //
+  // +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Kind string `json:"kind"`
+}
+```
+
+### Routes
+
+On Routes, we remove the `RouteGateways` struct and replace it with a list of
+parent references to attach to.
+
+#### Removed
+From Route Specs:
+```go
+  // Gateways defines which Gateways can use this Route.
+  //
+  // +optional
+  // +kubebuilder:default={allow: "SameNamespace"}
+  Gateways *RouteGateways `json:"gateways,omitempty"`
+```
+
+And the structs that references:
+```go
+// RouteGateways defines which Gateways will be able to use a route. If this
+// field results in preventing the selection of a Route by a Gateway, an
+// "Admitted" condition with a status of false must be set for the Gateway on
+// that Route.
+type RouteGateways struct {
+  // Allow indicates which Gateways will be allowed to use this route.
+  // Possible values are:
+  // * All: Gateways in any namespace can use this route.
+  // * FromList: Only Gateways specified in GatewayRefs may use this route.
+  // * SameNamespace: Only Gateways in the same namespace may use this route.
+  //
+  // +optional
+  // +kubebuilder:validation:Enum=All;FromList;SameNamespace
+  // +kubebuilder:default=SameNamespace
+  Allow *GatewayAllowType `json:"allow,omitempty"`
+
+  // GatewayRefs must be specified when Allow is set to "FromList". In that
+  // case, only Gateways referenced in this list will be allowed to use this
+  // route. This field is ignored for other values of "Allow".
+  //
+  // +optional
+  GatewayRefs []GatewayReference `json:"gatewayRefs,omitempty"`
+}
+```
+
+#### Added
+To Route Specs:
+```go
+  // ParentRefs references the parent resources this Route should be attached
+  // to. The only kind of parent resource with "Core" support is Gateway. This
+  // API may be extended in the future to support additional kinds of parent
+  // resources such as Routes. It is invalid to reference the same parent
+  // resource more than once.
+  //
+  // +optional
+  // +kubebuilder:validation:MaxItems=16
+  ParentRefs []ParentRef `json:"parentRefs,omitempty"`
+```
+
+And the struct that references:
+```go
+// ParentRef identifies an API object that should be considered a parent of this
+// resource. The only kind of parent resource with "Core" support is Gateway.
+// This API may be extended in the future to support additional kinds of parent
+// resources such as Routes.
+type ParentRef struct {
+  // Group is the group of the referent.
+  //
+  // Support: Core
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +kubebuilder:default=gateway.networking.k8s.io
+  Group string `json:"group"`
+
+  // Kind is kind of the referent.
+  //
+  // Support: Core (Gateway)
+  // Support: Extended (Other Resources)
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +kubebuilder:default=Gateway
```

Comment:
optional and pointer; can be taken care of in the implementation PR

Reply:
Updated, thanks!